src/shared_library.cc

```diff
@@ -107,6 +111,16 @@ SharedLibrary::OpenLibraryHandle(const std::string& path, void** handle)
 {
   LOG_VERBOSE(1) << "OpenLibraryHandle: " << path;
 
+#ifdef TRITON_ENABLE_GPU
+  // This call is to prevent a deadlock issue with dlopening backend shared
+  // libraries and calling CUDA APIs in other threads. Since CUDA API also
+  // dlopens some libraries and dlopen has an internal lock, it can create a
+  // deadlock. Intentionally ignore the CUDA_ERROR (if any) for containers
+  // running on a CPU.
+  int device_count;
+  cudaGetDeviceCount(&device_count);
+#endif
```

Comment:
I think you need to rebase with `main`.

Reply:
This should be resolved when `rmccormick-cache-api` -> `rmccormick-cache-main`